File: 프로그래머스/2/181188. 요격 시스템/요격 시스템.py
def solution(targets):
    targets.sort(key=lambda x: x[1])
    cnt = 0
    before = 0 # 초기화
    for s, e in targets:
        if s >= before: # 새로운 요격 필요 else에서 이 요격으로 격추할 수 있는 것들 포함한다
            cnt += 1
            before = e
    return cnt

File: 프로그래머스/2/181188. 요격 시스템/요격 시스템.java
import java.util.*;

class Solution {
    public int solution(int[][] targets) {
        
        int before = 0;
        int cnt = 0;
        Arrays.sort(targets, (o1, o2) -> o1[1] - o2[1]);
        
        for(int [] target: targets) {
            if (target[0] >= before){
                before = target[1];
                cnt++;
            }
        }
        
        return cnt;
    }
}